### DISCUSSION TITLE
Update user metadata using Zitadel API

### DISCUSSION THREAD

**Discussion by althaf004** [zitadel/zitadel#7064] [Q&A] [answered]
I had followed the instructions provided in this url: https://zitadel.com/docs/guides/integrate/private-key-jwt

i have created a service user . To generate the access token. used private key and using rs256 algorithm i was able to generate a jwt token. Using this token, a new access_token was generated using scopes as `openid email profile urn:zitadel:iam:user:metadata` and `grant_type=urn:ietf:params:oauth:grant-type:jwt-bearer`

1. when calling for userinfo, im getting only sub. 

```
{
  "sub": "242730346209083395"
}
```

Im trying to add a meta data to this service user. Please let me know how and which API should i follow to make it happen.

**Comment by hifabienne** [comment]
@muhlemmer @livio-a can you help here?

**Comment by muhlemmer** [comment, marked as answer]
Metadata can be added to the user here in the UI:

![image](https://github.com/zitadel/zitadel/assets/5411563/9d8aa76f-861e-4033-bca6-7f5a092318d4)

On the API, [Set User Metadata](https://zitadel.com/docs/apis/resources/mgmt/management-service-set-user-metadata).

Note for the scopes `email profile` you will not get anything back. A service user simply does not have `email` or `profile` information, so none will be returned.
